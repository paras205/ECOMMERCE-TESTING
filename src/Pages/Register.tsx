import { useForm } from "react-hook-form";

const Register = () => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  const onSubmit = handleSubmit((data) => console.log(data));
  return (
    <div>
      <form onSubmit={onSubmit}>
        <input
          {...register("email", { required: true, maxLength: 10 })}
          placeholder="email"
        />
        {errors?.email && <p>Enter a valid email</p>}
        <button type="submit">Register User</button>
      </form>
    </div>
  );
};

export default Register;
