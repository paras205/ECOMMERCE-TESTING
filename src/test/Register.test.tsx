import Register from "src/Pages/Register";
import renderer from "react-test-renderer";
import { fireEvent } from "@testing-library/react";
import { render, screen, act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("register", () => {
  it("register page renders correctly", () => {
    const tree = renderer.create(<Register />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("register user correctly", async () => {
    render(<Register />);
    const value = "user@test.com";
    const input = screen.getByPlaceholderText(/email/i);
    await userEvent.type(input, value);
    await act(async () => {
      fireEvent.click(screen.getByText(/Register User/i));
    });
  });
});
