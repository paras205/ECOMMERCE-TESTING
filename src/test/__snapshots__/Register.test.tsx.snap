// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register register page renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="email"
    />
    <button
      type="submit"
    >
      Register User
    </button>
  </form>
</div>
`;
